File: VB/App_Code/SampleData.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Web

Namespace Sample
    Public NotInheritable Class SampleData

        Private Sub New()
        End Sub


        Private Shared data_Renamed As DataTable
        Public Shared ReadOnly Property Data() As DataTable
            Get
                If data_Renamed Is Nothing Then
                    data_Renamed = GenerateSampleData()
                End If
                Return data_Renamed
            End Get
        End Property

        Private Shared Function GenerateSampleData() As DataTable
            data_Renamed = New DataTable()
            Dim keyColumn = data_Renamed.Columns.Add("Id")
            data_Renamed.Columns.Add("Name")
            data_Renamed.PrimaryKey = New DataColumn() { keyColumn }

            data_Renamed.Rows.Add(New Object() { "10115", "Augusta Delono" })
            data_Renamed.Rows.Add(New Object() { "10501", "Berry Dafoe" })
            data_Renamed.Rows.Add(New Object() { "10709", "Chris Cadwell" })
            data_Renamed.Rows.Add(New Object() { "10356", "Esta Mangold" })
            data_Renamed.Rows.Add(New Object() { "10401", "Frank Diamond" })
            data_Renamed.Rows.Add(New Object() { "10202", "Liam Bell" })
            data_Renamed.Rows.Add(New Object() { "10205", "Simon Newman" })
            data_Renamed.Rows.Add(New Object() { "10403", "Wendy Underwood" })

            Return data_Renamed
        End Function

        Public Shared Function Lookup(ByVal key As String) As String
            Dim row = Data.Rows.Find(key)
            Return If(row IsNot Nothing, row("Name").ToString(), "")
        End Function
    End Class
End Namespace

File: CS/App_Code/SampleData.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Web;

namespace Sample {
    public static class SampleData {
        static DataTable data;
        public static DataTable Data {
            get {
                if (data == null)
                    data = GenerateSampleData();
                return data;
            }
        }

        static DataTable GenerateSampleData() {
            data = new DataTable();
            var keyColumn = data.Columns.Add("Id");
            data.Columns.Add("Name");
            data.PrimaryKey = new DataColumn[] { keyColumn };

            data.Rows.Add(new object[] { "10115", "Augusta Delono" });
            data.Rows.Add(new object[] { "10501", "Berry Dafoe" });
            data.Rows.Add(new object[] { "10709", "Chris Cadwell" });
            data.Rows.Add(new object[] { "10356", "Esta Mangold" });
            data.Rows.Add(new object[] { "10401", "Frank Diamond" });
            data.Rows.Add(new object[] { "10202", "Liam Bell" });
            data.Rows.Add(new object[] { "10205", "Simon Newman" });
            data.Rows.Add(new object[] { "10403", "Wendy Underwood" });

            return data;
        }

        public static string Lookup(string key) {
            var row = Data.Rows.Find(key);
            return row != null ? row["Name"].ToString() : "";
        }
    }
}